### PR TITLE
feat: add momento vector index as a vector store provider

### PR DESCRIPTION
**Pull request by malandis**
**Description**:

- Added Momento Vector Index (MVI) as a vector store provider. This includes an implementation with docstrings, integration tests, a notebook, and documentation on the docs pages.
- Updated the Momento dependency in pyproject.toml and the lock file to enable access to MVI.
- Refactored the Momento cache and chat history session store to prefer using "MOMENTO_API_KEY" over "MOMENTO_AUTH_TOKEN" for consistency with MVI. This change is backwards compatible with the previous "auth_token" variable usage. Updated the code and tests accordingly.

**Dependencies**:

- Updated Momento dependency in pyproject.toml.

**Testing**:

- Run the integration tests with a Momento API key. Get one at the [Momento Console](https://console.gomomento.com) for free. MVI is available in AWS us-west-2 with a superuser key.
- `MOMENTO_API_KEY=<your key> poetry run pytest tests/integration_tests/vectorstores/test_momento_vector_index.py`

**Tag maintainer:**

@eyurtsev

**Twitter handle**:

Please mention @momentohq for this addition to langchain. With the integration of Momento Vector Index, Momento caching, and session store, Momento provides serverless support for the core langchain data needs.

Also mention @mlonml for the integration.

### CODE REVIEW

**Comment by malandis** [comment]
I rebased and solved merge conflicts in [ded2e78](https://github.com/langchain-ai/langchain/pull/11567/commits/ded2e78b59632351905f4c48e08e477e7d102957)

File: libs/langchain/langchain/vectorstores/momento_vector_index.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+                    VectorIndexConfigurations.Default.latest(),
+                    credential_provider=CredentialProvider.from_environment_variable(
+                        "MOMENTO_API_KEY"
+                    ),
+                ),
+                index_name="my-index",
+            )
+    """
+
+    _client: "PreviewVectorIndexClient"
```
[the remark this answers is not in the document]
Comment (malandis) [reply libs/langchain/langchain/vectorstores/momento_vector_index.py:56]: I started this in the middle of the vector store schema refactor. `MomentoVectorIndex`, the vector store class, inherits from `langchain.schema.vectorstore.VectorStore`. Is this not the appropriate base type?

[the remark this answers is not in the document]
Comment (malandis) [reply libs/langchain/langchain/vectorstores/momento_vector_index.py:56]: No it wasn't. Good catch!

[the remark this answers is not in the document]
Comment (malandis) [reply libs/langchain/langchain/vectorstores/momento_vector_index.py]: Fixed in https://github.com/langchain-ai/langchain/pull/11567/commits/d9010a6cafc94129ab9238d1362c1197604fe921

**Comment by baskaryan** [comment]
lgtm, thanks @malandis!

**Comment by malandis** [comment]
Thanks @baskaryan and @hwchase17 🙇 . Any chance @momentohq can get a Twitter shoutout for the contribution so we can reshare also 🙏 ?

**Pull request merged**